code/game/objects/structures/mannequin.dm

```diff
@@ -0,0 +1,1187 @@
+
+/////////////////////////////////////////////////////////MANNEQUINS//////////////////////////////////////////////////////////
+//Basically statues that you can dress up.
+
+#define MANNEQUIN_ICONS_SLOT		"slot_icon"
+#define MANNEQUIN_ICONS_PRIMITIVE	"primitive_icon"
+#define MANNEQUIN_ICONS_SPECIES		"species_icon"
+#define MANNEQUIN_ICONS_FAT			"fat_icon"
+#define MANNEQUIN_DYNAMIC_LAYER		"dynamic_layer"
+
+/obj/structure/mannequin
+	name = "human marble mannequin"
+	desc = "You almost feel like it's going to come alive any second."
+	icon = 'icons/obj/mannequin.dmi'
+	icon_state="mannequin_marble_human"
+	density = 1
+	anchored = 1
+	plane = ABOVE_HUMAN_PLANE
+	layer = VEHICLE_LAYER
+	var/datum/species/species
+	var/species_type = /datum/species/human //TODO: mannequin for other races
+	var/fat = 0
+	var/primitive = 0
+	var/list/clothing = list()
+	var/list/obj/item/held_items = list(null, null)
+	var/clothing_offset_x = 0
+	var/clothing_offset_y = 3*PIXEL_MULTIPLIER
+	var/health = 90
+	var/maxHealth = 90
+
+	var/list/all_slot_icons = list()
+
+	//for mappers, with love
+	var/mapping_uniform = null
+	var/mapping_shoes = null
+	var/mapping_gloves = null
+	var/mapping_ears = null
+	var/mapping_suit = null
+	var/mapping_glasses = null
+	var/mapping_hat = null
+	var/mapping_belt = null
+	var/mapping_mask = null
+	var/mapping_back = null
+	var/mapping_id = null
+
+	var/mapping_hand_right = null
+	var/mapping_hand_left = null
+
+/obj/structure/mannequin/New()
+	..()
+
+	species = new species_type()
+
+	clothing = list(
+		SLOT_MANNEQUIN_ICLOTHING,
+		SLOT_MANNEQUIN_FEET,
+		SLOT_MANNEQUIN_GLOVES,
+		SLOT_MANNEQUIN_EARS,
+		SLOT_MANNEQUIN_OCLOTHING,
+		SLOT_MANNEQUIN_EYES,
+		SLOT_MANNEQUIN_BELT,
+		SLOT_MANNEQUIN_MASK,
+		SLOT_MANNEQUIN_HEAD,
+		SLOT_MANNEQUIN_BACK,
+		SLOT_MANNEQUIN_ID,
+		)
+	get_all_slot_icons()
+	checkMappingWear()
+
+
+/obj/structure/mannequin/Destroy()
+	for(var/cloth in clothing)
+		if(clothing[cloth])
+			var/obj/item/cloth_to_drop = clothing[cloth]
+			cloth_to_drop.forceMove(loc)
+			clothing[cloth] = null
+	for(var/item in held_items)
+		if(held_items[item])
+			var/obj/item/item_to_drop = held_items[item]
+			item_to_drop.forceMove(loc)
+			held_items[item] = null
+	..()
+
+
+/obj/structure/mannequin/MouseDrop(var/mob/M)
+	..()
+	if(M != usr)
```

Comment:
This probably needs a lot more sanity, mousedrop/click other procs like that are the least sane you could possibly be